File: cloud-code/SpendCurrency.js
// SpendCurrency Cloud Code Function
const { EconomyApi } = require("@unity-services/economy-1.0");

module.exports = async ({ params, context, logger }) => {
  try {
    const { currencyId, amount } = params;
    if (!currencyId || !amount) throw new Error("Missing required parameters");
    if (amount <= 0) throw new Error("Amount must be positive");

    const balance = await EconomyApi.getCurrencyBalance({ currencyId });
    if (balance.amount < amount) throw new Error("Insufficient funds");

    await EconomyApi.spendCurrency({ currencyId, amount });
    logger.info(`Spent ${amount} ${currencyId} from player`);

    return { success: true, currencyId, amount, newBalance: balance.amount - amount };
  } catch (error) {
    logger.error(`SpendCurrency failed: ${error.message}`);
    throw error;
  }
};

File: cloud-code-csharp/src/SpendCurrency.cs
using Unity.Services.CloudCode;
using Unity.Services.Economy;
using Unity.Services.Core;
using System;
using System.Threading.Tasks;

[CloudCodeFunction("SpendCurrency")]
public class SpendCurrency : CloudCodeFunction
{
    public async Task<object> ExecuteAsync(CloudCodeRequest request)
    {
        try
        {
            // Parse parameters
            var currencyId = request.GetParameter<string>("currencyId");
            var amount = request.GetParameter<int>("amount");

            // Validate parameters
            if (string.IsNullOrEmpty(currencyId))
                throw new ArgumentException("currencyId is required");
            
            if (amount <= 0)
                throw new ArgumentException("amount must be positive");

            // Get current balance
            var balance = await EconomyService.Instance.Currency.GetCurrencyBalanceAsync(currencyId);
            
            if (balance.Amount < amount)
                throw new InvalidOperationException("Insufficient funds");

            // Spend currency
            await EconomyService.Instance.Currency.SpendCurrencyAsync(currencyId, amount);

            // Log success
            Logger.LogInfo($"Spent {amount} {currencyId} from player");

            return new
            {
                success = true,
                currencyId = currencyId,
                amount = amount,
                newBalance = balance.Amount - amount
            };
        }
        catch (Exception ex)
        {
            Logger.LogError($"SpendCurrency failed: {ex.Message}");
            throw;
        }
    }
}